### ISSUE TITLE
No way to get upstream back out of EasyFramed

### ISSUE THREAD

**Issue by spinda** [tokio-rs/tokio-core#73]
Ideally `EasyFramed` would contain something like the following:

``` rust
    pub fn into_upstream(self) -> T {
        self.upstream
    }
```

I find myself needing this for a protocol that switches between two different transport formats during a single session (authentication -> main data exchange).

I can submit this as a quick PR if it'd be welcome.

**Comment by carllerche** [comment]
Sure, a PR would be good. Following rust std conventions, there should probably be three functions for accessing the inner value:
- `get_ref`
- `get_mut`
- `into_inner`

**Comment by alexcrichton** [comment]
Sounds good to me!

**Comment by alexcrichton** [comment]
Fixed in #76
